source/sigv4_http_request.c

```diff
@@ -342,10 +342,12 @@ static void s_aws_signing_waiter_clean_up(struct aws_signing_waiter *waiter) {
     aws_condition_variable_clean_up(&waiter->signal);
 }
 
-void s_sign_callback(struct aws_signing_result *result, void *user_data) {
+void s_sign_callback(struct aws_signing_result *result, int error_code, void *user_data) {
     struct aws_signing_waiter *waiter = user_data;
     aws_mutex_lock(&waiter->lock);
 
+    AWS_FATAL_ASSERT(error_code == AWS_ERROR_SUCCESS);
```

Comment:
so this can't ever be called when error_code is set?

Reply:
Oooh good catch, I thought this was a unit test lol